internal/loki/ingester/wal/encoding.go

```diff
@@ -0,0 +1,251 @@
+package wal
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/prometheus/prometheus/tsdb/chunks"
+	"github.com/prometheus/prometheus/tsdb/record"
+
+	"github.com/grafana/loki/v3/pkg/logproto"
+	"github.com/grafana/loki/v3/pkg/util/encoding"
+)
+
+// RecordType represents the type of the WAL/Checkpoint record.
+type RecordType byte
+
+const (
+	_ = iota // ignore first value so the zero value doesn't look like a record type.
+	// WALRecordSeries is the type for the WAL record for series.
+	WALRecordSeries RecordType = iota
+	// WALRecordEntriesV1 is the type for the WAL record for samples.
+	WALRecordEntriesV1
+	// CheckpointRecord is the type for the Checkpoint record based on protos.
+	CheckpointRecord
+	// WALRecordEntriesV2 is the type for the WAL record for samples with an
+	// additional counter value for use in replaying without the ordering constraint.
+	WALRecordEntriesV2
+	// WALRecordEntriesV3 is the type for the WAL record for samples with structured metadata.
+	WALRecordEntriesV3
+)
+
+// The current type of Entries that this distribution writes.
+// Loki can read in a backwards compatible manner, but will write the newest variant.
+// TODO: Change to WALRecordEntriesV3?
```

Comment:
Any reason to keep this TODO?

Reply:
I'll remove it. I suppose this was missed when [the update to v3 was done](https://github.com/grafana/loki/commit/287f29bc46e077f43cfa8778732bdcb43ed882a3#diff-1f70440a13bd620fe23bf8e7d0e2f181b435a5baa9d3655249d237f22daaf2fc).